### ISSUE TITLE
`xml-editor`: Improve Table Format Editor interface

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#4370]
The 'Table Format Editor' displays the same information in both applications, just in different ways. I believe the Specify 6 implementation has a more intuitive user interface and experience when configuring these.

It is super easy to use and allows you to quickly switch between a table view for easily viewing what format will be used for each field value and a single table format which just lets you pick from a list of fields.

https://github.com/specify/specify7/assets/37256050/8cb519e6-958d-49eb-860d-0c55bb503771

Conditional or "switch" table formats are not exceedingly common but are used in cases like agents and reference work. The interface for these can be tricky, but I believe Specify. 6 does a good job.

### Specify 6

It is very intuitive. To start, it is clear for the user that they need to pick a type between two options. a "Single table format" or "Display depends on value of field:" and an easy way to pick which field that is. 

<img width="532" alt="image" src="https://github.com/specify/specify7/assets/37256050/de1f05df-b513-4d87-83ee-7b3932d14c1c">

I can simply see that there are several possible values for that field, and the format is clear to see based on those values:

| Field Value | Table Format |
|----------|----------|
| 0   | `lastName`  |
| 1   | `lastName` `middleInitial` `lastName`  |
| 2   | `lastName`  |
| 3   | `lastName`  |

### Specify 7

There is a very long list of every possibility separated by a full-width line containing the field value that results in the configured format. 

Besides the Christmas-themed formatting, it is clear it could use the grid view that Specify 6 has above. Imagine that, and just like in Specify 6, you could click on a line and open another dialog with the fields and separators contained within.

https://github.com/specify/specify7/assets/37256050/84a5b0e7-4c97-4cb6-b3ae-41b0de87d3bd

<img width="473" alt="image" src="https://github.com/specify/specify7/assets/37256050/5436f3a9-df6f-4a96-81d8-6338c9e4325e">

### Specify 7
<img width="206" alt="Screenshot 2024-01-08 at 10 41 41 AM" src="https://github.com/specify/specify7/assets/37256050/e9d51564-d292-434d-a929-e9b9dc918602">

### Specify 6
<img width="648" alt="image" src="https://github.com/specify/specify7/assets/37256050/317d67f3-c878-4d21-a07f-5fe7ff0e94b9">

## TL;DR

Specify 6 has a great table format editor that we should take inspiration from and match more closely in the future.

**Comment by grantfitzsimmons** [comment]
I also think semantically 'Formatters' is less clear than 'Table Formats'.

Maybe it's worth considering a relabel to 'Table Formats' and 'Table Aggregators'
